File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        HashMap<String, Integer> map = new HashMap<>();
        for(int i=0; i<phone_book.length; i++){
            map.put(phone_book[i], i);
        }
        
        for(int i=0; i<phone_book.length; i++){
            String checker = phone_book[i];
            for(int j=0; j<checker.length(); j++){
                if(map.containsKey(checker.substring(0,j))) return false;       
            }
        }
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):      
    dictionary = {}
    for i in range(len(phone_book)):
        dictionary[phone_book[i]] = i
        
    for i in dictionary.keys():
        for j in range(len(i)):
            if i[:j] in dictionary.keys():
                return False
    return True